OracleGoldenGate/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# LICENSE CDDL 1.0 + GPL 2.0
+#
+# Copyright (c) 2014-2017 Oracle and/or its affiliates. All rights reserved.
+#
+# ORACLE GOLDENGATE DOCKER PROJECT
+# --------------------------------
+#
+# Dockerfile for Oracle GoldenGate
+#
+# REQUIRED FILES TO BUILD THIS IMAGE
+# ----------------------------------
+#
+# From http://www.oracle.com/technetwork/middleware/goldengate/downloads/index.html
+#  Download the file for "Oracle GoldenGate 12.x.x.x.x for Oracle on Linux x86-64"
+#    - fbo_ggs_Linux_x64_shiphome.zip
+#
+# HOW TO BUILD THIS IMAGE
+# -----------------------
+#
+# Execute the "buildDockerImage.sh" shell script, with the name of the downloaded OGG filename.
+# Example:
+#      $ ./buildDockerImage.sh ~/Downloads/fbo_ggs_Linux_x64_shiphome.zip
+#
+
+ARG         BASE_IMAGE=container-registry.oracle.com/database/instantclient:12.2.0.1
```

Comment:
Since Oracle GoldenGate is delivered for a wide variety of databases, it's likely that the end-user will want to use a base image other than the default I have here.  For example, Oracle GoldenGate for MySQL doesn't need the Oracle Instant Client base image, which is what I am using for the default.

In order to allow a different base image to be used, a parameter can be used in the FROM statement.  But, in order to use variables in FROM, they *must* be declared first using an ARG statement.  See https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact for additional information.